### PR TITLE
Implement destructuring binders for pairs

### PR DESCRIPTION
**Pull request by MonoidMusician**
Now we can bind pairs `(p , q)` in all places where binders appear (lambdas, sigmas, pis, lets).

I took the liberty of updating most of the `.poly` files to use this style, it's a big improvement.

(Needs to be squash merged)

### CODE REVIEW

File: examples/prelude.poly [hunk 1]
```diff
@@ -24,8 +24,8 @@ def dup : Π (A : Type), A → (A × A) :=
 def pair : ℕ → ℕ → ℕ × ℕ :=
   λ n m → (n , m : ℕ)
 
-def projl : (ℕ × ℕ) → ℕ :=
-  λ x → fst x
+def projl : Π (T : Type), (ℕ × T) → T :=
+  λ T (_ , x) → x
```
Comment (solomon-b) [inline examples/prelude.poly:28]:
```suggestion
def projr : Π (T : Type), (ℕ × T) → T :=
  λ T (_ , x) → x
```

File: examples/prelude.poly [hunk 2]
```diff
@@ -113,9 +114,8 @@ def test-pair : nyan ⇒ tensor nyan nyan :=
 #normalize (test-pair 1)
 
 def swap : Π (P Q : Poly), tensor P Q ⇒ tensor Q P :=
-  λ P Q → λ pq⁺ pq⁻ ⇝
-    let⁻ (p⁻ , q⁻) := pq⁻;
-    return (snd pq⁺ , fst pq⁺) ⇜ (q⁻ , p⁻)
+  λ P Q → λ (p⁺ , q⁺) (p⁻ , q⁻) ⇝
+    return (q⁺ , p⁺) ⇜ (q⁻ , p⁻)
```
Comment (solomon-b) [inline examples/prelude.poly:118]: :100:

**Review by solomon-b — APPROVED** [review]

**Comment by MonoidMusician** [comment]
there's some bug with levels, `Tutorial.poly` and `Prelude.poly` both fail with `bad do_fst`, can't tell when I introduced it

**Comment by MonoidMusician** [comment]
Okay, I'll merge this soon if there are no objections.

**Pull request merged**